packages/react/src/components/Checkbox/Checkbox.js

```diff
@@ -54,6 +60,14 @@ const Checkbox = React.forwardRef(function Checkbox(
             ref.current = el;
           }
         }}
+        // readonly attribute not applicable to type="checkbox"
+        // see - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox
+        aria-readonly={readOnly}
+        onClick={(evt) => {
+          if (readOnly) {
+            evt.preventDefault();
+          }
+        }}
```

Comment:
Is there any potential use case where click should still be fired, even if it's readonly? Maybe the user needs to be shown a notification or something else to indicate why it's readonly?

Reply:
Agreed. I believe I assumed `preventDefault` would not stop propagation.

Added test for onClick and passed on if the user supplies a method.

Reply:
Note: onClick is normally passed through from read-only inputs.